### ISSUE TITLE
onHeartbeat error

### ISSUE THREAD

**Issue by wvries** [Xenomes/Domoticz-TinyTUYA-Plugin#13]
Hello,

I have the following error when I start up:

2023-02-06 16:11:34.128 Error: Tuya: handleThread: string indices must be integers
This occurs only once when restarting Domoticz or restarting the plugin.

and then every 10 seconds an error regarding the Heartbeat.
2023-02-06 16:11:40.078 Error: Tuya: Call to function 'onHeartbeat' failed, exception details:
2023-02-06 16:11:40.083 Error: Tuya: Traceback (most recent call last):
2023-02-06 16:11:40.083 Error: Tuya: File "/home/pi/domoticz/plugins/Domoticz-TinyTUYA-Plugin/plugin.py", line 321, in onHeartbeat
2023-02-06 16:11:40.083 Error: Tuya: _plugin.onHeartbeat()
2023-02-06 16:11:40.084 Error: Tuya: File "/home/pi/domoticz/plugins/Domoticz-TinyTUYA-Plugin/plugin.py", line 282, in onHeartbeat
2023-02-06 16:11:40.084 Error: Tuya: if time.time() - last_update < 60 and testData == False:
2023-02-06 16:11:40.084 Error: Tuya: NameError: name 'last_update' is not defined

Running Domoticz 2022.2 (build 15006)
Python V3.9.2
OS "Raspbian GNU/Linux 11 (bullseye)"
HW: RPI3+

**Comment by Xenomes** [comment]
Hi,
I think you have a light that is not detected correctly, the script is bypassed by that. Will correct it in the next version.
Is there in Domoticz a text device created?

**Comment by wvries** [comment]
Hi,

no text devices created.
In the Tuya platform, all devices are online.

**Comment by Xenomes** [comment]
Can you provide a dump.json so i can test you devices? You can create it by edit and run the debug_discovery.py from the tools directory.

**Comment by varazir** [comment]
I have the same issue and I updated the plugin today

**Comment by wvries** [comment]
Hi, I edited and ran the debug_discovery.py but got a message debug_discovery: -1 line 44 and a debug.json that is empty.
Looks like the devices cannot be found (although all existing and online in the cloud and in the app - and working - on the phone).
double checked the API key, secret and device ID... all correct

EDIT:
run in debug mode.
2023-02-09 15:06:52.076 Tuya: ----> 'devs' '{'Error': 'Error Response from Tuya Cloud', 'Err': '913', 'Payload': "Error from Tuya Cloud: Code 28841002: 'No permissions. Your subscription to cloud development plan has expired.'"}'

Seems my subscription has expired.

Requested extension of the IOT API subscription and will test afterwards and update ticket.

**Comment by pfolcato** [comment]
Hi, you can extend the trial period by 6 months on this page... but it's quite boring having to do it every time...
![immagine](https://user-images.githubusercontent.com/74878466/217896634-a6535e9e-4cb0-4207-ad8c-a956e2ebd5d9.png)

**Comment by wvries** [comment]
Yep, done the extension for 6 moths and now it is all working again.

Only error I get is when restarting the plugin, it tries to create a device that is already there.
I will open another ticket for this.

For the subscription expire, maybe a catch of this error in the plugin to check if the subscription is still active.

Will also open another ticket for this and closing this one.

**Comment by varazir** [comment]
> Hi, you can extend the trial period by 6 months on this page... but it's quite boring having to do it every time... ![immagine](https://user-images.githubusercontent.com/74878466/217896634-a6535e9e-4cb0-4207-ad8c-a956e2ebd5d9.png)

Could you share the link to the page,  have been clickingaround for 5 min and can't find exact that page

**Comment by varazir** [comment]
Found it but maybe suggest send some other error when this happen så other knows why.
